### DISCUSSION TITLE
difference between img_size and patch_size in UNETR

### DISCUSSION THREAD

**Discussion by elahe-badali** [Project-MONAI/MONAI#2748] [Q&A] [answered]
Hi @ahatamiz,
Thank you for sharing your code. I have a question about it. what is difference between img_size and patch_size in arguments of UNETR. In MONAI documentation example it is mentioned that "img_size (Union[Sequence[int], int]) – dimension of input image." but in example:

for single channel input 4-channel output with patch size of (96,96,96), feature size of 32 and batch norm
>>> net = UNETR(in_channels=1, out_channels=4, img_size=(96,96,96), feature_size=32, norm_name='batch')

 it said that (96,96,96) is patch size but assigned it to img_size.
I think it's a little bit confusing.

**Comment by ahatamiz** [comment, marked as answer]
Hi @elahe-badali 

Thanks for your interest in our work. `img_size` represents the dimension of input image (e.g. sampled `(96,96,96)` patches of input CT images) whereas `patch_size` is the size of input patches (e.g. `(16,16,16)` ) used in the encoder of UNETR (i.e. ViT).

Thanks

**Reply by elahe-badali** [reply]
Hi again @ahatamiz ,
Thanks for your answer. I want to run your framework with my cardiac MRI data. Each image has shape of (64,64,25). How can i set img_size and patch_size for my data?

**Comment by LalehSeyyed** [comment]
Hi 
I have CT images of size (512,512, z) where z varies between 40 to a much larger number (say 400) depends on the patient. I am wondering how we should set up img_size when they do not all have a fixed size in z?

**Comment by ahatamiz** [comment]
Hi @LalehSeyyed, `img_size` will then be the same as the predefined size that is used for sampling the data (e.g. (96,96,96) for BTCV). It is a hyper-parameter which needs tuning for each specific task. For brain tumor segmentation (BRATS) for instance, we set it to (128,128,128) to better capture tumors of variable sizes.
